pkg/smartcontract/binding/generate.go

```diff
@@ -19,34 +19,48 @@ import (
 )
 
 const srcTmpl = `
-{{- define "METHOD" -}}
-// {{.Name}} {{.Comment}}
-func {{.Name}}({{range $index, $arg := .Arguments -}}
-	{{- if ne $index 0}}, {{end}}
-		{{- .Name}} {{.Type}}
-	{{- end}}) {{if .ReturnType }}{{ .ReturnType }} {
-	return neogointernal.CallWithToken(Hash, "{{ .NameABI }}", int(contract.{{ .CallFlag }})
-		{{- range $arg := .Arguments -}}, {{.Name}}{{end}}).({{ .ReturnType }})
-	{{- else -}} {
-	neogointernal.CallWithTokenNoRet(Hash, "{{ .NameABI }}", int(contract.{{ .CallFlag }})
-		{{- range $arg := .Arguments -}}, {{.Name}}{{end}})
-	{{- end}}
-}
-{{- end -}}
-// Code generated by neo-go contract generate-wrapper --manifest <file.json> --out <file.go> --hash <hash> [--config <config>]; DO NOT EDIT.
+// Code generated by neo-go contract generate-wrapper --manifest <file.json> --out <file.go> [--hash <hash>] [--config <config>]; DO NOT EDIT.
 
 // Package {{.PackageName}} contains wrappers for {{.ContractName}} contract.
 package {{.PackageName}}
 
 import (
 {{range $m := .Imports}}	"{{ $m }}"
 {{end}})
-
+{{if .Hash}}
 // Hash contains contract hash in big-endian form.
-const Hash = "{{ .Hash }}"
+var Hash = "{{ .Hash }}"
+{{end}}
+
+// Contract represents a smart contract with a dynamic hash.
+type Contract struct {
+    hash interop.Hash160
+}
+
+// NewContract creates a new instance of Contract with the given hash.
+func NewContract(hash interop.Hash160) *Contract {
+    return &Contract{hash: hash}
+}
+
 {{range $m := .Methods}}
 {{template "METHOD" $m }}
-{{end}}`
+{{end}}
+
+{{- define "METHOD" -}}
+// {{.Name}} {{.Comment}}
+func (c *Contract) {{.Name}}({{range $index, $arg := .Arguments -}}
+    {{- if ne $index 0}}, {{end}}
+        {{- .Name}} {{.Type}}
+    {{- end}}) {{if .ReturnType }}({{ .ReturnType }}, error) {
+    return contract.Call(c.hash, "{{ .NameABI }}", contract.{{ .CallFlag }}
+        {{- range $arg := .Arguments -}}, {{.Name}}{{end}})
+    {{- else -}}error {
+    return contract.Call(c.hash, "{{ .NameABI }}", contract.{{ .CallFlag }}
```

Comment:
Should be `neogointernal.CallWithToken*` if hash is provided. Wen need to keep it as an optimisation.